Sources/Core/AWSSDKHTTPAuth/Sources/AWSSDKHTTPAuth/AWSSigV4Signer.swift

```diff
@@ -256,12 +256,14 @@ public class AWSSigV4Signer: SmithyHTTPAuthAPI.Signer {
             }
         }
 
```

Comment:
By this point in this method, we know that the stream is going to be sent to server using chunked encoding. Therefore, regardless of checksum presence, if body is unsigned body, we have to resolve to `.streamingUnsignedPayloadTrailer`.